rust/tests/ffi.rs

```diff
@@ -0,0 +1,184 @@
+#![allow(clippy::undocumented_unsafe_blocks, unused_crate_dependencies)]
```

Comment:
This is not possible because we want to be able to pass raw pointers which could be dereferenced. see [clippy](https://rust-lang.github.io/rust-clippy/master/index.html#/not_unsafe_ptr_arg_deref)

Reply:
I now found a way to make it safe. There are now two wrapper functions. One takes a reference which makes it safe and the other one takes no context and passes a null pointer to the *_raw functions.